### ISSUE TITLE
`FATAL:  postmaster became multithreaded during startup` on macOS 10.12

### ISSUE THREAD

**Issue by jakob** [PostgresApp/PostgresApp#643]
We have discovered that running PostgreSQL 13 and 14 on macOS 10.12 (Sierra) is no longer possible since Postgres.app 2.5

We saw log messages like this:
> FATAL:  postmaster became multithreaded during startup
> HINT:  Set the LC_ALL environment variable to a valid locale.

The hint was a red herring - the problem is not related to locale at all.

We have discovered that the problem started when we compiled with OpenSSL 1.1.1l -- when using 1.1.1k everything works.

We are planning to downgrade to OpenSSL 1.1.1k for now, and we hope the issue will be resolved in OpenSSL 1.1.1m

**Comment by jakob** [comment]
@tbussmann has unfortunately confirmed that the issue persists in the current OpenSSL 1.1.1m-dev snapshot.

**Comment by tbussmann** [comment]
I manually reviewed the code changes from openssl 1.1.1k to 1.1.1l for suspect changes. With this I found the inclusion of `CommonCrypto/CommonRandom.h` and use of `CCRandomGenerateBytes` in `/crypto/rand/rand_unix.c`. Testing 1.1.1l with that change reverted no longer shows the `postmaster became multithreaded during startup` error on macOS 10.12!

While I don't yet understand why this call seems to use threads on 10.12 but not on 10.13 onwards, the recent commit history of openssl shows this change is causing more trouble: the header was missing in older OS X versions, [preventing openssl to build,](https://github.com/openssl/openssl/issues/16407) so [it's use was limited](https://github.com/openssl/openssl/commit/96ac8f13f4d0ee96baf5724d9f96c44c34b8606c) to OS X >= 10.10 - later, OS X 10.10 and 10.11 [exhibited problems with this function](https://github.com/openssl/openssl/issues/16517) as well, so the use [was limited further](https://github.com/openssl/openssl/commit/2f3b120401533db82e99ed28de5fc8aab1b76b33) to macOS >=10.12.

I now see two possible ways for us: either patch `include/crypto/rand.h` in our builds of openssl >= 1.1.1m to limit the definition of `OPENSSL_APPLE_CRYPTO_RANDOM` to macOS >= 10.13, or try to convince upstream to do so. While waiting for 1.1.1m we could also backport the before mentioned commits into 1.1.1l and use that.
